packages/flutter/lib/src/widgets/resizing_header_sliver.dart

```diff
@@ -0,0 +1,213 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'slotted_render_object_widget.dart';
+
+/// A sliver that is pinned to the start of its [CustomScrollView] and
+/// reacts to scrolling by resizing between the intrinsic sizes of its
+/// min and max extent prototypes.
+///
+/// The minimum and maximum sizes of this sliver are defined by [minExtentPrototype]
+/// and [maxExtentPrototype], a pair of widgets that are laid out once. You can
+/// use [SizedBox] widgets to define the size limits.
+///
+/// This sliver is preferable to the general purpose [SliverPersistentHeader]
+/// for its relatively narrow use case because there's no need to create a
+/// [SliverPersistentHeaderDelegate] or to predict the header's minimum or
+/// maximum size.
+///
+/// {@tool dartpad}
+/// This sample shows how this sliver's two extent prototype properties can be used to
+/// create a resizing header whose minimum and maximum sizes match small and large
+/// configurations of the same header widget.
+///
+/// ** See code in examples/api/lib/widgets/sliver/resizing_header_sliver.0.dart **
+/// {@end-tool}
```

Comment:
Can we add a todo to add `See also`s to all these new slivers that can be combined together after they all land?

Reply:
Yes.